File: Python3/프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    person1 = [1, 2, 3, 4, 5, 1, 2, 3, 4, 5]
    person2 = [2, 1, 2, 3, 2, 4, 2, 5, 2, 1, 2, 3, 2, 4, 2, 5]
    person3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5, 3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    
    cnt1 = 0
    cnt2 = 0
    cnt3 = 0
    
    for i in range(len(answers)):
        if person1[i%len(person1)] == answers[i]:
            cnt1 += 1
        if person2[i%len(person2)] == answers[i]:
            cnt2 += 1
        if person3[i%len(person3)] == answers[i]:
            cnt3 += 1
            
    max_score = max(cnt1, cnt2, cnt3)
    
    if cnt1 == max_score:
        answer.append(1)
    if cnt2 == max_score:
        answer.append(2)
    if cnt3 == max_score:
        answer.append(3)
    
    return answer

File: C++/프로그래머스/1/42840. 모의고사/모의고사.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> persons;
    vector<int> person1 = {1,2,3,4,5};
    vector<int> person2 = {2,1,2,3,2,4,2,5};
    vector<int> person3 = {3,3,1,1,2,2,4,4,5,5};
    
    int answer1 = 0;
    int answer2 = 0;
    int answer3 = 0;
    
    // 각 수포자의 답을 독립적으로 검사
    for (int i = 0; i < answers.size(); i++) {
        if (answers[i] == person1[i % person1.size()]) {
            answer1++;
        }
        if (answers[i] == person2[i % person2.size()]) {
            answer2++;
        }
        if (answers[i] == person3[i % person3.size()]) {
            answer3++;
        }
    }

    // 가장 많이 맞춘 사람을 찾음
    int maxScore = max({answer1, answer2, answer3});
    
    // 가장 많이 맞춘 사람을 persons 벡터에 넣음
    if (answer1 == maxScore) persons.push_back(1);
    if (answer2 == maxScore) persons.push_back(2);
    if (answer3 == maxScore) persons.push_back(3);
    
    return persons;
}
